### PR TITLE
[ADAM-1242] Clean up dead code in org.bdgenomics.adam.util.

### LINKED ISSUE #1242: Dead-ish code cleanup in `org.bdgenomics.adam.utils`
- [ ] [MapTools](https://github.com/bigdatagenomics/adam/blob/ca4d9649bac2523f9951c24f2e184a1b37f9ba06/adam-core/src/main/scala/org/bdgenomics/adam/util/MapTools.scala) is imported in one class, but unused
- [ ] [RegExp](https://github.com/bigdatagenomics/adam/blob/master/adam-core/src/main/scala/org/bdgenomics/adam/util/RegExp.scala) is used in one class but should be removed
- [ ] [IntervalListReader](https://github.com/bigdatagenomics/adam/blob/master/adam-core/src/main/scala/org/bdgenomics/adam/util/IntervalListReader.scala) is unused
- [ ] [ImplicitJavaConversions](https://github.com/bigdatagenomics/adam/blob/master/adam-core/src/main/scala/org/bdgenomics/adam/util/ImplicitJavaConversions.scala) is imported in four classes but should be removed
- [ ] [Util](https://github.com/bigdatagenomics/adam/blob/master/adam-core/src/main/scala/org/bdgenomics/adam/util/Util.scala) is used in one place but should be removed

### PR DESCRIPTION
**Pull request by fnothaft**
Resolves #1242. Removed:

* org.bdgenomics.adam.util.ImplicitJavaConversions
* org.bdgenomics.adam.util.IntervalListReader
* org.bdgenomics.adam.util.MapTools
* org.bdgenomics.adam.util.RegExp
* org.bdgenomics.adam.util.Util

### CODE REVIEW

**Comment by fnothaft** [comment]
I've added two commits:

* 694dbe2 does a general documentation cleanup
* f148b07 moves several classes/objects to different packages, and tightens up protections on these classes

This is good to go from my side.

File: adam-core/src/main/scala/org/bdgenomics/adam/util/ReferenceContigMap.scala [hunk 1]
```diff
@@ -17,15 +17,26 @@
...
+ * A broadcastable ReferenceFile backed by a map containing contig name ->
+ * Seq[NucleotideContigFragment] pairs.
+ *
+ * @param contigMap a map containing a Seq of contig fragments per contig.
+ */
 case class ReferenceContigMap(contigMap: Map[String, Seq[NucleotideContigFragment]]) extends ReferenceFile {
 
-  // create sequence dictionary
+  /**
+   * The sequence dictionary corresponding to the contigs in this two bit file.
```
Comment (heuermh) [inline adam-core/src/main/scala/org/bdgenomics/adam/util/ReferenceContigMap.scala]: this isn't a two bit file

**Comment by heuermh** [comment]
Looks great other than one doc fix

**Comment by fnothaft** [comment]
Fixed the documentation issue and pushed.

**Comment by heuermh** [comment]
Thanks, @fnothaft!

**Pull request merged**